Modules/Devices/src/Devices.Application/Identities/Queries/GetRegisteredIdentityDetails/Handler.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Domain.Entities.Identities;
+using MediatR;
+
+namespace Backbone.Modules.Devices.Application.Identities.Queries.GetRegisteredIdentityDetails;
+public class Handler : IRequestHandler<GetRegisteredIdentityDetailsQuery, GetRegisteredIdentityDetailsResponse>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly IUserContext _userContext;
+
+    public Handler(IIdentitiesRepository identitiesRepository, IUserContext userContext)
+    {
+        _identitiesRepository = identitiesRepository;
+        _userContext = userContext;
+    }
+
+    public async Task<GetRegisteredIdentityDetailsResponse> Handle(GetRegisteredIdentityDetailsQuery request, CancellationToken cancellationToken)
+    {
+        var identity = await _identitiesRepository.FindByAddress(_userContext.GetAddress(), cancellationToken) ?? throw new NotFoundException(nameof(Identity));
```

Comment:
You shouldn't throw a NotFoundException here. If there is nothing coming back from the repository, this means that either the JWT is invalid, or that the data of the identity is lost. Both are not the fault of the user and should therefore result in a 500 instead of a 4xx.
Throw a normal exception with a meaningful error message.

Reply:
Changes made.